packages/components/src/modifiers/hds-float-popover.js

```diff
@@ -0,0 +1,155 @@
+import { modifier } from 'ember-modifier';
+
+import {
+  arrow,
+  autoUpdate,
+  computePosition,
+  flip,
+  // this could be used in the future if we want to give consumers an option to hide the popover when the "trigger" hides from the viewport
+  // see: https://floating-ui.com/docs/hide
+  // hide,
+  offset,
+  shift,
+  // this could be used in the future if we want to give consumers an option to:
+  // - let the "popover" auto-resize when there is not enough space (usually vertical) in the viewport to contain the entire popover
+  // - let the "popover" match the width of the "trigger" (it may have min/max width/heigh via CSS too)
+  // see: https://floating-ui.com/docs/size
+  // notice: below you can find a preliminary code implementation that was tested and worked relatively well
+  // size,
+} from '@floating-ui/dom';
+
+// we use this function to process all the options provided to the modifier in a single place,
+// in relation to the Floating UI APIs, and keep the modifier code more clean/simple
+function getFloatingUIOptions(popoverOptions) {
+  let {
+    popoverPlacement = 'bottom',
+    popoverPositionStrategy = 'absolute', // we don't need to use `fixed` if we use the Popover API (it puts the element in the `top-layer`)
+    popoverOffsetOptions,
+    popoverEnableCollisionDetection,
+    popoverFlipOptions = { padding: 8 },
+    popoverShiftOptions = { padding: 8 },
+    popoverArrow,
+    popoverMiddlewareExtra = [],
+  } = popoverOptions;
+
+  // we build dynamically the list of middleware functions to invoke, depending on the options provided
+
+  const popoverMiddleware = [];
+
+  // https://floating-ui.com/docs/offset
+  popoverMiddleware.push(offset(popoverOffsetOptions));
+
+  // TODO? split the option in two to support collision detection only in one direction like for the power-select?
```

Comment:
For the PowerSelect, by default, the dropdown list matches the "trigger" width. In that case, the ShiftOptions probably don't make sense to use.

However, if matchTriggerWidth is set to "false" (meaning that the dropdown list could be wider than the trigger), then it could make sense to shift the position of the dropdown to keep it in view. I only found one example of this option in use in HashiCorp projects when I searched previously, however.

So it could make sense to split these options into two...

Reply:
The `enableCollisionDetection` now supports a boolean (`true`) as well as a string (`flip` or `shift`) to enable distinct behaviours.